### DISCUSSION TITLE
analyzing large datasets

### DISCUSSION THREAD

**Discussion by stamb123** [satijalab/seurat#5418] [Q&A]
We have 388,000 cells for SCA. We are running into limitations with R with memory.  Is there a workaround so we can cluster all 388,000 cells in a single Seurat object?

**Comment by samuel-marsh** [comment]
Hi,

Can you paste the error that you are getting and what the RAM available on your system is?

Best,
Sam

**Comment by stamb123** [comment]
Sam:

Thanks for your assistance.  Below is the error message that my bioinformatician, Nico,  received.  Any assistance would be appreciated. We are trying to cluster 245,000 cells post filtering.

Dwight Stambolian

From: Dana, Nicholas ***@***.***>
Date: Wednesday, January 5, 2022 at 9:23 AM
To: Stambolian, Dwight ***@***.***>
Subject: Re: [External] Re: [satijalab/seurat] analyzing large datasets (Discussion #5418)
Sure,

The error below occurred during the IntegrateData function, and there is 128 GB of RAM on the system.

...
Integrating data
Merging dataset 3 27 29 11 12 10 1 28 9 2 30 19 13 18 15 17 into 32 31 34 33 39 40 41 42 4 37 36 38 35 6 8
Extracting anchors for merged samples
Finding integration vectors
Error: vector memory exhausted (limit reached?)

Nico

________________________________
From: Stambolian, Dwight ***@***.***>
Sent: Saturday, December 25, 2021 1:22 PM
To: Dana, Nicholas ***@***.***>
Subject: FW: [External] Re: [satijalab/seurat] analyzing large datasets (Discussion #5418)


Can you answer the below?



Dwight



From: Samuel Marsh ***@***.***>
Reply-To: satijalab/seurat ***@***.***>
Date: Saturday, December 25, 2021 at 9:48 AM
To: satijalab/seurat ***@***.***>
Cc: "Stambolian, Dwight" ***@***.***>, Author ***@***.***>
Subject: [External] Re: [satijalab/seurat] analyzing large datasets (Discussion #5418)



Hi,

Can you paste the error that you are getting and what the RAM available on your system is?

Best,
Sam

—
Reply to this email directly, view it on GitHub<https://github.com/satijalab/seurat/discussions/5418#discussioncomment-1869817>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AXATY4AW2DZCZYDWYOLPSVLUSXKTNANCNFSM5KTMR6LQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Comment by stamb123** [comment]
Hi Sam:

I am resending my response to your query. I also cannot find my question in the Seurat discussion board.  Was it deleted?


Can you help us?

Thanks
Dwight Stambolian


Sam:

Thanks for your assistance.  Below is the error message that my bioinformatician, Nico,  received.  Any assistance would be appreciated. We are trying to cluster 245,000 cells post filtering.

Dwight Stambolian

From: Dana, Nicholas ***@***.***>
Date: Wednesday, January 5, 2022 at 9:23 AM
To: Stambolian, Dwight ***@***.***>
Subject: Re: [External] Re: [satijalab/seurat] analyzing large datasets (Discussion #5418)
Sure,

The error below occurred during the IntegrateData function, and there is 128 GB of RAM on the system.

...
Integrating data
Merging dataset 3 27 29 11 12 10 1 28 9 2 30 19 13 18 15 17 into 32 31 34 33 39 40 41 42 4 37 36 38 35 6 8
Extracting anchors for merged samples
Finding integration vectors
Error: vector memory exhausted (limit reached?)

Nico

________________________________
From: Stambolian, Dwight ***@***.***>
Sent: Saturday, December 25, 2021 1:22 PM
To: Dana, Nicholas ***@***.***>
Subject: FW: [External] Re: [satijalab/seurat] analyzing large datasets (Discussion #5418)


Can you answer the below?



Dwight



From: Samuel Marsh ***@***.***>
Reply-To: satijalab/seurat ***@***.***>
Date: Saturday, December 25, 2021 at 9:48 AM
To: satijalab/seurat ***@***.***>
Cc: "Stambolian, Dwight" ***@***.***>, Author ***@***.***>
Subject: [External] Re: [satijalab/seurat] analyzing large datasets (Discussion #5418)



Hi,

Can you paste the error that you are getting and what the RAM available on your system is?

Best,
Sam

—
Reply to this email directly, view it on GitHub<https://github.com/satijalab/seurat/discussions/5418#discussioncomment-1869817>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AXATY4AW2DZCZYDWYOLPSVLUSXKTNANCNFSM5KTMR6LQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Comment by timoast** [comment]
@stamb123 the error indicates there wasn't enough RAM. I'd suggest using the `IntegrateEmbeddings()` function rather than `IntegrateData()`, or reduce the number of features being integrated
